includes/theme-support/class-wc-unsupported-theme-product-single.php

```diff
@@ -0,0 +1,98 @@
+<?php
+/**
+ * Unsupported theme single product page enhancements.
+ *
+ * @package  WooCommerce/Classes
+ * @since   3.3.0
+ * @version 3.3.0
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * Single product pages support for themes that don't declare WooCommerce support.
+ *
+ * @class   WC_Unsupported_Theme_Product_Single
+ * @package WooCommerce/Classes
+ * @since   3.3.0
+ */
+class WC_Unsupported_Theme_Product_Single {
+
+	/**
+	 * Used for marking whether we're in the content filter.
+	 *
+	 * @var bool
+	 */
+	protected static $in_content_filter = false;
+
+	/**
+	 * Theme enhancements init.
+	 */
+	public static function init() {
+		remove_action( 'woocommerce_before_main_content', 'woocommerce_output_content_wrapper', 10 );
+		remove_action( 'woocommerce_after_main_content', 'woocommerce_output_content_wrapper_end', 10 );
+		remove_filter( 'template_include', array( 'WC_Template_Loader', 'template_loader' ) );
+		remove_filter( 'comments_template', array( 'WC_Template_Loader', 'comments_template_loader' ) );
+
+		add_theme_support( 'wc-product-gallery-zoom' );
```

Comment:
If we're enabling by default we need to ensure we add this inside our new theming docs.